src/index.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * Created by meganschoendorf on 7/13/16.
+ */
+
+/**
+ * This decorator calculates the time it takes the method to execute and calls the function sendDuration with the class
+ * name, function name, and duration in ms.
+ *
+ * This decorator should only be used on methods that return a promise as the function sendDuration is called in a
+ * .then clause. If your method does not return a promise, use the decorator timed.
+ * @param sendDuration
+ * @returns {function(any, string, TypedPropertyDescriptor<Function>): undefined}
+ */
+export interface extraDataDogTags {
+    category?: string;
+}
+
+//   declare type MethodDecorator = <T>(target: Object, propertyKey: string | symbol, descriptor: TypedPropertyDescriptor<T>) => TypedPropertyDescriptor<T> | void;
+
+export function timedAsync(
+    sendDuration: (className: string, functionName: string, durationMs: number) => Promise<any>
+): MethodDecorator {
+    return (
+        target: Object,
+        propertyName: string | symbol,
+        descriptor: TypedPropertyDescriptor<any>
+    ) => {
+        let functionName: string = String(propertyName);
+
+        const className = target.constructor.name;
+        let method = descriptor.value;
+        descriptor.value = function() {
+            let start = new Date();
+            if (!method) {
+                throw new Error('method missing');
```

Comment:
do we want to throw in here? could make a function fail that we wouldn't want to fail due to datadog, and i don't know how to wrap an annotation in a try catch.

Reply:
tbh, i didnt want to break anything so left it alone.